packages/react-native-client/src/debug/internal/eventPayloadValidator.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  ReceivableEventPayloads,
+  ReceivableEvents,
+} from '../../RNFishjamClientModule';
+
+import type {
+  GenericMetadata,
+  SimulcastConfig,
+  TrackMetadata,
+} from '../../types';
+import type {
+  AudioOutputDevice,
+  AudioOutputDeviceType,
+  OnAudioDeviceEvent,
+} from '../../hooks/useAudioSettings';
+import type {
+  Camera,
+  CameraFacingDirection,
+  CurrentCameraChangedType,
+} from '../../hooks/useCamera';
+import type { AudioTrack, Peer, VideoTrack } from '../../hooks/usePeers';
+import type { ReconnectionStatus } from '../../hooks/useConnection';
+import type { PeerStatus } from '../../hooks/useConnection';
+
+export function isNativeEventPayloadValid<
```

Comment:
WDYT about using [zod](https://zod.dev) here? Even just for fun.

Reply:
Added!